openpype/hosts/fusion/plugins/publish/collect_render.py

```diff
@@ -0,0 +1,210 @@
+        instances_to_remove = []
+
+        current_file = context.data["currentFile"]
+        version = context.data["version"]
+
+        project_entity = context.data["projectEntity"]
+
+        for inst in context:
+            if not inst.data.get("active", True):
+                continue
+
+            family = inst.data["family"]
+            if family != "render":
+                continue
+
+            task_name = inst.data.get("task")  # legacy
+            tool = inst.data["transientData"]["tool"]
+
+            instance_families = inst.data.get("families", [])
+            subset_name = inst.data["subset"]
+            instance = FusionRenderInstance(
+                family="render",
+                tool=tool,
+                workfileComp=comp,
+                families=instance_families,
+                version=version,
+                time="",
+                source=current_file,
+                label="{} - {}".format(subset_name, family),
+                subset=subset_name,
+                asset=inst.data["asset"],
+                task=task_name,
+                attachTo=False,
+                setMembers='',
+                publish=True,
+                name=subset_name,
+                resolutionWidth=comp_frame_format_prefs.get("Width"),
+                resolutionHeight=comp_frame_format_prefs.get("Height"),
+                pixelAspect=aspect_x / aspect_y,
+                tileRendering=False,
+                tilesX=0,
+                tilesY=0,
+                review="review" in instance_families,
+                frameStart=inst.data["frameStart"],
+                frameEnd=inst.data["frameEnd"],
+                handleStart=inst.data["handleStart"],
+                handleEnd=inst.data["handleEnd"],
+                ignoreFrameHandleCheck=(not inst.data.get("custom_range")),
+                frameStep=1,
+                fps=comp_frame_format_prefs.get("Rate"),
+                app_version=comp.GetApp().Version,
+                publish_attributes=inst.data.get("publish_attributes", {})
+            )
+
+            render_target = inst.data["creator_attributes"]["render_target"]
+
+            if render_target == "local":
+                # for local renders
+                self._instance_data_local_update(
+                    project_entity, instance, f"render.{render_target}")
+
+            if render_target == "frames":
+                self._instance_data_local_update(
+                    project_entity, instance, f"render.{render_target}")
+
+            if render_target == "farm":
+                fam = "render.farm"
+                if fam not in instance.families:
+                    instance.families.append(fam)
+                instance.toBeRenderedOn = "deadline"
+                instance.farm = True  # to skip integrate
+                if "review" in instance.families:
+                    # to skip ExtractReview locally
+                    instance.families.remove("review")
```

Comment:
Why not for `farm` also add the `render.farm` family so the [`FusionSubmitDeadline` wouldn't need the `instance.data["farm"]` check](https://github.com/ynput/OpenPype/pull/4955/files#diff-ccca8f8c11192f70e98b240acb8097031db3bc7436f82834009f759688f45756R78-R81) but just runs on `render.farm` family instead?

Reply:
This is the limitation of the current publisher -  if we want to see attributes from `FusionSubmitDeadline` then this has to be `render` family. The `render.farm` family will be added during collecting and at the moment of setting up those attributes in publisher the instance is still only `render` family. If I will add `render.farm` family to the deadline plugin, its attributes will disapear from the publishers right pane.